src/v/compaction/utils.cc

```diff
@@ -29,10 +39,10 @@ bool is_removable_control_batch(
     // Fence batches can also be immediately removed when seen in the
     // `__consumer_offsets` topic or safely removed from a user topic. However,
     // removal in a user topic is gated by
-    // `log_compaction_disable_tx_batch_removal()`.
+    // `log_compaction_tx_batch_removal_enabled()`.
     auto is_co_topic = model::is_consumer_offsets_topic(ntp);
     auto remove_user_tx_fence_enabled
-      = !config::shard_local_cfg().log_compaction_disable_tx_batch_removal()
+      = config::shard_local_cfg().log_compaction_tx_batch_removal_enabled()
```

Comment:
This change should be in the previous commit. Otherwise it won't build: we access deprecated config value, which is of type `ss::sstring`, and attempt to apply `!` operator to it. That's how I test all commits can be built: `vtools/scripts/crawl-down.sh bazel build  --//src/v/redpanda:lto=False -j 28 --config=stamp //...` (takes quite a while).

Reply:
I don't quite have a vision on our strategy regarding what goes into which repo. Is there a summary?